src/components/molecules/WalletNetworkSwither.tsx

```diff
@@ -0,0 +1,67 @@
+import React, { ReactElement, useState, useEffect } from 'react'
+import { useWeb3 } from '../../providers/Web3'
+import { addCustomNetwork, getNetworkConfigObject } from '../../utils/web3'
+import Button from '../atoms/Button'
+import styles from './WalletNetworkSwitcher.module.css'
+import useNetworkMetadata from '../../hooks/useNetworkMetadata'
+import NetworkName from '../atoms/NetworkName'
+import { getOceanConfig } from '../../utils/ocean'
+import { useAsset } from '../../providers/Asset'
+
+export default function WalletNetworkSwitcher(): ReactElement {
+  const { networkId, web3Provider } = useWeb3()
+  const { networksList } = useNetworkMetadata()
+  const { ddo } = useAsset()
+  const DEFAULT_ETH_CHAIN_IDS = [1, 3, 4]
```

Comment:
they should come from `appConfig.chainIds` which hold the default selected networks

```
const { appConfig } = useSiteMetadata()
```

Reply:
I think the name is confusing. There I wanted to keep track of the built in Metamask network to know when to show the switch button

Reply:
ah right, now I get it, thanks! In other news, we might not have to do that anymore, as https://github.com/MetaMask/metamask-extension/pull/11184 seems finally be shipped in latest MetaMask version. I'll test a bit with other dapps and will report back

Reply:
Oh good news, looking forward for that

Reply:
so the new Balancer v2 frontend does this where using their network switcher also switches wallet network and it works for ETH Mainnet:

<img width="1359" alt="Screen Shot 2021-07-12 at 16 55 08" src="https://user-images.githubusercontent.com/90316/125309281-eb914180-e331-11eb-8fb5-f01c07115142.png">

And MetaMask docs now have this handy code snippet showing how to combine `wallet_addEthereumChain` with `wallet_switchEthereumChain` so we do not have to maintain a list of builtin networks:

https://docs.metamask.io/guide/rpc-api.html#usage-with-wallet-switchethereumchain

Reply:
Cool, it's working :)